nakama/cardinal.go

```diff
@@ -61,27 +65,35 @@ func cardinalListEndpoints(path string) ([]string, error) {
 		return nil, fmt.Errorf("list endpoints (at %q) failed with status code %d: %v", url, resp.StatusCode, string(buf))
 	}
 	dec := json.NewDecoder(resp.Body)
-	var endpoints []string
-	if err := dec.Decode(&endpoints); err != nil {
+	var endpointsStruct endpoints
+	if err := dec.Decode(&endpointsStruct); err != nil {
 		return nil, err
 	}
-	return endpoints, nil
+	return &endpointsStruct, nil
+}
 
+func cardinalListQueryEndpoints() ([]string, error) {
+	endpointsStruct, err := cardinalGetEndpointsStruct()
+	if err != nil {
+		return nil, err
+	}
+	return endpointsStruct.QueryEndpoints, nil
 }
 
-func cardinalListAllEndpoints() ([]string, error) {
-	var endpoints []string
-	txs, err := cardinalListEndpoints(listTxEndpointsEndpoint)
+func cardinalListTxEndpoints() ([]string, error) {
+	endpointsStruct, err := cardinalGetEndpointsStruct()
 	if err != nil {
 		return nil, err
 	}
-	endpoints = append(endpoints, txs...)
-	reads, err := cardinalListEndpoints(listReadEndpoints)
+	return endpointsStruct.QueryEndpoints, nil
+}
+
+func cardinalListAllEndpoints() ([]string, error) {
```

Comment:
This isn't actually used, so it can be removed.

Reply:
Why? it could be used in the future. I don't see the harm in leaving it.

Reply:
As a general rule we should remove unused code. This specific instances won't be harmful, but in aggregate, unused code will contribute to the cognitive load to understand the project, make refactoring more difficult, and increase the size of the project binary.